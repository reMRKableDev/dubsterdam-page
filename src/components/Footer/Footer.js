import React from "react";
import Emoji from "../Emoji/Emoji";
import {
  StyledFooter,
  StyledH2,
  StyledFooterPhrase,
  StyledFooterDeveloperTag,
} from "./FooterStyles";

const Footer = () => {
  return (
    <StyledFooter>
      <StyledH2>
        <span>Thank You</span>
      </StyledH2>
      <StyledFooterPhrase>for checking us out!</StyledFooterPhrase>
      <StyledFooterDeveloperTag>
        Made with <Emoji symbol="❤️" label="love" /> by{" "}
        <a href="https://remrkabledev.com/">reMRKable Dev</a>
      </StyledFooterDeveloperTag>
    </StyledFooter>
  );
};

export default Footer;
