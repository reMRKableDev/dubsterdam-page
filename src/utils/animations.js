import { keyframes } from "styled-components";
import { bounce, fadeInUp, fadeInDown, shake, merge } from "react-animations";

export const bounceAnimation = keyframes`${bounce}`;
export const fadeInUpAnimation = keyframes`${fadeInUp}`;
export const fadeInDownAnimation = keyframes`${fadeInDown}`;
export const shakeAnimation = keyframes`${shake}`;
export const fadeInUpShakeAnimation = keyframes`${merge(fadeInUp, shake)}`;
