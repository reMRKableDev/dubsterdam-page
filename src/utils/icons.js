import { Mail, Github, Linkedin, Database } from "@styled-icons/feather";
import {
  Css3,
  Html5,
  ReactLogo,
  Nodejs,
  Javascript,
} from "@styled-icons/boxicons-logos";

export const mailIcon = Mail;
export const gitHubIcon = Github;
export const linkedInIcon = Linkedin;
export const databaseIcon = Database;
export const cssIcon = Css3;
export const htmlIcon = Html5;
export const reactIcon = ReactLogo;
export const nodeIcon = Nodejs;
export const javascriptIcon = Javascript;
