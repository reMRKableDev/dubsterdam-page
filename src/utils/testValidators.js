export const validateNotEmpty = (received) => {
  expect(received).not.toBeNull();
  expect(received).toBeTruthy();
};

export const validateDarkThemeMode = (received) => {
  expect(received).not.toMatch(/light/i);
  expect(received).toMatch(/dark/i);
};

export const validateLightThemeMode = (received) => {
  expect(received).not.toMatch(/dark/i);
  expect(received).toMatch(/light/i);
};
