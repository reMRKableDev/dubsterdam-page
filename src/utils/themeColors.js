import theme from "styled-theming";

export const bgColor = theme("mode", {
  light: "#fafafa",
  dark: "#222",
});

export const footerColor = theme("mode", {
  light: "#222",
  dark: "#fafafa",
});

export const textColor = theme("mode", {
  light: "#000",
  dark: "#fff",
});

export const buttonBgColor = theme("mode", {
  light: "#222",
  dark: "#eee",
});

export const buttonTextColor = theme("mode", {
  light: "#eee",
  dark: "#222",
});

export const brandColor = theme("mode", {
  light: "#ed6c6d",
  dark: "#ed6c6d",
});

export const textAccentColor = theme("mode", {
  light: "#555",
  dark: "#fff",
});
