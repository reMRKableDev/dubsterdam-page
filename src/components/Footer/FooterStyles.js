import styled from "styled-components";
import { footerColor, brandColor, bgColor } from "../../utils/themeColors";
import { respondTo } from "../../layout/respondTo";

export const StyledFooter = styled.footer`
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 300px;
  background: ${footerColor};
  z-index: -1;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const StyledH2 = styled.h2`
  font-size: 6vw;
  font-family: "Rubik", sans-serif;
  margin: 1em 0;
  text-transform: uppercase;

  span {
    color: ${brandColor};
    border: 4px solid ${brandColor};
    box-shadow: 0 0 10px ${!bgColor};
    padding: 5px;
    text-transform: none;
  }

  ${respondTo.sm`
    font-size: 5.5vw;
    margin: 0.8em 0;

    span {
        border: 6px solid ${brandColor};
    }
  `}

  ${respondTo.md`
    font-size: 5vw;
    margin: 0.6em 0;

    span {
        border: 8px solid ${brandColor};
    }
  `}
  
  ${respondTo.lg`
    font-size: 4.5vw;
    margin: 0.5em 0;

    span {
        border: 10px solid ${brandColor};
    }
  `}
`;

export const StyledFooterPhrase = styled.p`
  font-family: "Karla", sans-serif;
  color: ${bgColor};
  margin-bottom: 25px;
`;

export const StyledFooterDeveloperTag = styled.p`
  font-family: "Rubik", sans-serif;
  color: ${bgColor};

  a {
    color: ${brandColor};
    text-decoration: none;
    cursor: pointer;
  }
`;
