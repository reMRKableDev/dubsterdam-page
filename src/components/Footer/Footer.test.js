import React from "react";
import { shallow } from "enzyme";
import Footer from "./Footer";

import { validateMatchSnapshot } from "../../utils/testValidators";

describe("Footer component unit tests", () => {
  it("should shallow render component and match snapshot", () => {
    const shallowComponent = () => shallow(<Footer />);
    const wrapper = shallowComponent();

    validateMatchSnapshot(wrapper);
  });
});
