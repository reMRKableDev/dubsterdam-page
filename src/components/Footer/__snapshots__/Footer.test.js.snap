// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component unit tests should shallow render component and match snapshot 1`] = `
<styled.footer>
  <styled.h2>
    <span>
      Thank You
    </span>
  </styled.h2>
  <styled.p>
    for checking us out!
  </styled.p>
  <styled.p>
    Made with 
    <Emoji
      label="love"
      symbol="❤️"
    />
     by
     
    <a
      href="https://remrkabledev.com/"
    >
      reMRKable Dev
    </a>
  </styled.p>
</styled.footer>
`;
